pkg/webhook/managedresource/validatingadmissionpolicy.go

```diff
@@ -14,63 +14,74 @@ const resourceName = "aks-fleet-managed-by-arm"
 
 var forbidden = metav1.StatusReasonForbidden
 
-func GetValidatingAdmissionPolicy(isHub bool) *admv1.ValidatingAdmissionPolicy {
+func getValidatingAdmissionPolicy(isHub bool) *admv1.ValidatingAdmissionPolicy {
 	vap := &admv1.ValidatingAdmissionPolicy{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "admissionregistration.k8s.io/v1",
-			Kind:       "ValidatingAdmissionPolicy",
-		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name: resourceName,
 		},
-		Spec: admv1.ValidatingAdmissionPolicySpec{
-			MatchConstraints: &admv1.MatchResources{
-				ObjectSelector: &metav1.LabelSelector{
-					MatchLabels: map[string]string{
-						"fleet.azure.com/managed-by": "arm",
-					},
+	}
+	mutateValidatingAdmissionPolicy(vap, isHub)
+	return vap
+}
+
+func mutateValidatingAdmissionPolicy(vap *admv1.ValidatingAdmissionPolicy, isHub bool) {
+	vap.TypeMeta = metav1.TypeMeta{
+		APIVersion: "admissionregistration.k8s.io/v1",
+		Kind:       "ValidatingAdmissionPolicy",
+	}
+	vap.ObjectMeta = metav1.ObjectMeta{
```

Comment:
You probably need to preserve the ResourceVersion

Reply:
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.22.1/pkg/controller/controllerutil/controllerutil.go#L320

Example scenaio of an update I can think of is member agent going down and coming back up,

If the user updated the object and when we rollback to what we expect the object to look like via an update we will only preserve the resourceVersion of the update the user made ?

@jim-minter is there another reason to preserve resource version I'm missing ?

Reply:
The resource version comes from etcd, not anywhere else.  It increments every time the object is changed and is used for optimistic concurrency control.  It's necessary for it to be maintained through get->mutate->update otherwise the update call will be rejected.